src/server/debugcmd.cc

```diff
@@ -306,29 +350,55 @@ void DebugCmd::Populate(CmdArgList args) {
   (*cntx_)->SendOk();
 }
 
-void DebugCmd::PopulateRangeFiber(uint64_t from, uint64_t len, std::string_view prefix,
-                                  unsigned value_len, bool populate_random_values) {
+void DebugCmd::PopulateRangeFiber(uint64_t from, uint64_t num_of_keys,
+                                  const PopulateOptions& options) {
   ThisFiber::SetName("populate_range");
-  VLOG(1) << "PopulateRange: " << from << "-" << (from + len - 1);
+  VLOG(1) << "PopulateRange: " << from << "-" << (from + num_of_keys - 1);
 
-  string key = absl::StrCat(prefix, ":");
+  string key = absl::StrCat(options.prefix, ":");
   size_t prefsize = key.size();
   DbIndex db_indx = cntx_->db_index();
   EngineShardSet& ess = *shard_set;
   std::vector<PopulateBatch> ps(ess.size(), PopulateBatch{db_indx});
   SetCmd::SetParams params;
 
-  for (uint64_t i = from; i < from + len; ++i) {
-    StrAppend(&key, i);
+  uint64_t index = from;
+  uint64_t added = 0;
+  while (added < num_of_keys) {
+    if ((index > from) && (index % num_of_keys == 0)) {
+      index = index - num_of_keys + options.total_count;
+    }
+    StrAppend(&key, index);
+
+    if (options.slot_range.has_value()) {
+      // Each fiber will add num_of_keys. Key are in the form of <key_prefix>:<index>
+      // We need to make sure that different fibers will not add the same key.
+      // Fiber starting <key_prefix>:<from> to <key_prefix>:<num_of_keys-1>
+      // than continue to <key_prefix>:<total_count> to <key_prefix>:<total_count+num_of_keys-1>
+      // and continue untill num_of_keys are added.
+
+      // Add keys only in slot range.
+      SlotId sid = ClusterConfig::KeySlot(key);
+      if (sid < options.slot_range->start || sid > options.slot_range->end) {
+        key.resize(prefsize);  // shrink back
```

Comment:
I'd move this `resize()` (and the one below) to be right before `StrAppend(&key, index)` above, it feels less error prone (and repetitive).